pkg/apis/core/types.go

```diff
@@ -2430,6 +2461,14 @@ type ContainerStatus struct {
 	// +optional
 	ContainerID string
 	Started     *bool
+	// Node compute resources allocated for the container.
+	// +featureGate=InPlacePodVerticalScaling
+	// +optional
+	ResourcesAllocated ResourceList
```

Comment:
Do we want these to be statically defaulted (e.g. set == spec.resources) or left empty until kubelet sets them?

Reply:
Imo it is better we let the kubelet set it, as it better represents our definition that it stands for the [resources allocated on the node](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1287-in-place-update-pod-resources#api-changes) (by kubelet) for the container.

Reply:
ACK